File: Greedy/2437번 - 저울/2437.py
import sys

input = sys.stdin.readline

n = int(input())
coin = list(map(int, input().split()))

def solve():
    ans = 1
    sorted_coin = sorted(coin)
    
    for c in sorted_coin:
        if ans < c: 
            break
        ans += c
    
    print(ans)

if __name__ == '__main__':
    solve()

File: Greedy/2437번 - 저울/2437.cpp
#include <iostream>
#include <algorithm>

using namespace std;

#define ll long long
#define LIM 1001

int n;
int coin[LIM];
bool weight[LIM];

void input() {
    cin >> n;
    for(int i = 0, c; i < n; i++) {
        cin >> coin[i];
    }
}

void solve() {
    ll ans = 1;
    
    sort(coin, coin + n);
    
    for(int i = 0; i < n; i++) {
        if(coin[i] > ans) break;
        ans += coin[i];
    }
    
    cout << ans << '\n';
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}